engine.go

```diff
@@ -459,12 +462,20 @@ func (e *Engine) Start() error {
 
 	e.AddSources(typeSource, scopeSource, sourceSource)
 
+	e.prepCache()
+	e.cacheContext, e.cacheCancel = context.WithCancel(context.Background())
+
 	// Start purging cache
-	e.cache.StartPurger()
+	e.cache.StartPurger(e.cacheContext)
 
 	return e.connect()
 }
 
+// prepCache Prepares the cache for use
+func (e *Engine) prepCache() {
+	e.cache = sdpcache.NewCache()
```

Comment:
```suggestion
	if e.cache != nil {
		e.cache = sdpcache.NewCache()
	}
```
or should this go into a NewEngine() call that pushes this code to the owner of the Engine?

Reply:
I know, I thought about that too... It changes the whole way that you'd create engines which means that the sources would need to change, but it kind of needs it...

Reply:
Aaargh you're right I'm gonna do it